File: backends/php-laravel/app/Support/Dtos/AbstractDto.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\Support\Dtos;

use App\Support\Dto;
use App\Support\Traits\Assign;
use App\Support\Traits\Attribute;
use App\Support\Traits\Initialize;

/**
 * 基底DTOクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 * @package App\Support\Dtos
 */
abstract class AbstractDto implements Dto
{
    use Assign;
    use Attribute;
    use Initialize;

    /**
     * @var int|null 処理実行者ID
     */
    public ?int $executorId = null;

    /**
     * @var int|null バージョン
     */
    public ?int $version = null;

    /**
     * コンストラクタ
     */
    public function __construct()
    {
        $this->initializer();
    }
}

File: backends/java-spring/src/main/java/com/authorization/support/dtos/AbstractDto.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.support.dtos;

import lombok.Data;

/**
 * 基底DTOクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
public abstract class AbstractDto {

    private Long executorId;
    private Integer version;
}
